DogTales/dog/player.cpp

```diff
@@ -1,15 +1,28 @@
 #include <dog/player.hpp>
 
 namespace dog {
-Player::Player(glm::vec2 const world_space) : m_world_space(world_space) { m_sprite.set_size(size_v); }
+Player::Player(bave::App& app, glm::vec2 const world_space) : m_app(app), m_world_space(world_space) {
+	m_sprite.set_size(size_v);
+}
 
 void Player::tick(bave::Seconds const dt) {
 
 	m_physics.tick(dt);
 	m_sprite.transform.position = m_physics.position;
 
+	auto const& key_state = m_app.get_key_state();
+	auto direction = glm::vec2{};
+	if (key_state.is_pressed(bave::Key::eW) || key_state.is_pressed(bave::Key::eUp)) { direction.y += 1.0f; }
+	if (key_state.is_pressed(bave::Key::eS) || key_state.is_pressed(bave::Key::eDown)) { direction.y -= 1.0f; }
+	if (key_state.is_pressed(bave::Key::eA) || key_state.is_pressed(bave::Key::eLeft)) { direction.x -= 1.0f; }
+	if (key_state.is_pressed(bave::Key::eD) || key_state.is_pressed(bave::Key::eRight)) { direction.x += 1.0f; }
+
+	if (direction.x != 0.0f || direction.y != 0.0f) {
+		direction = glm::normalize(direction);
+		auto const displacement = direction * speed_v * dt.count();
+		m_physics.position += displacement;
```

Comment:
Two problems:
- This introduces a frame's delay between physics updating and the player's sprite catching up to it in the next tick. Set the sprite's position to physics position after physics position has been finalized.
- `handle_wall_collision` clamps the sprite's position, but this is never synchronized with physics. Either set `m_physics.position = m_sprite.transform.position` _after_ clamping the sprite position, or clamp the physics position directly.

Reply:
Made changes. Believe they're good.